Framework/LiveData/src/LoadLiveData.cpp

```diff
@@ -391,6 +391,32 @@ Workspace_sptr LoadLiveData::appendMatrixWSChunk(Workspace_sptr accumWS,
   return accumWS;
 }
 
+//----------------------------------------------------------------------------------------------
+/** Resets all HistogramX in given workspace to a single bin.
+ *
+ * Ensures bin boundaries encompass all events currently in the workspace.
+ * This will overwrite any rebinning that was previously.
+ *
+ * Input should be an EventWorkspace or WorkspaceGroup containing
+ * EventWorkspaces. Any other workspace types are ignored.
+ *
+ * @param workspace :: Workspace that will have its bins reset
+ */
+void LoadLiveData::resetAllXToSingleBin(API::Workspace_sptr workspace) {
```

Comment:
Minor point, but argument need not be a `shared_ptr<Workspace>` could just be a `Workspace &`. Just clears up the semantics a little since we are doing nothing with the shared_ptr aspects here. The code below could be changed to a regular `dynamic_cast` in that case.

Reply:
Would a `Workspace_sptr&` be more appropriate, since both `chunkWS` and `m_accumWS` are already `Workspace_sptr`? Or would `Workspace&` still be better here?